src/main/java/com/hotels/bdp/circustrain/bigquery/listener/BigQueryReplicationListener.java

```diff
@@ -39,9 +39,13 @@ public BigQueryReplicationListener(BigQueryDataExtractionManager dataExtractionM
   }
 
   @Override
-  public void tableReplicationStart(EventTableReplication tableReplication, String eventId) {
-    Table table = getTable(tableReplication);
-    dataExtractionManager.extract(table);
+  public void tableReplicationStart(EventTableReplication eventTableReplication, String eventId) {
+    log.info("Table replication from {}.{} -> {}.{} has started.",
+        eventTableReplication.getSourceTable().getDatabaseName(), eventTableReplication.getSourceTable().getTableName(),
+        eventTableReplication.getReplicaTable().getDatabaseName(),
+        eventTableReplication.getReplicaTable().getTableName());
+    Table table = getTable(eventTableReplication);
+    dataExtractionManager.register(table);
```

Comment:
I think you can assume in the lifecycle of 1 tableReplication start and either success or failure, will get called once. So just set the table into the extractionManager and unset on success/failure. This is not great but at least you can then get rid of the cache, the wrapper and the "if already extracted" mechanics. 
You should be safe to assume that the copier is called just once and the extraction on this table as result is also only done once. If you have reason to believe this is not the case, we need to have a deeper look.

Reply:
Metastore Client accesses the extraction location multiple times and this first occurs before the copier phase in CT. Basically operations follow the order getLocation -> extract -> cleanup, location needs to be generated first then referenced again multiple times in the metastore client and in extract and cleanup functionality. Got rid of the checks for repeated extractions + cleanups but we need to generate and hold onto the locations before the data extraction

Reply:
Replace Cache with single Table as we will only use one Table per replication anyway